### ISSUE TITLE
06.5 JOBSTER-API: showStats - ISSUE Node: 20.x - Mongoose version  "mongoose": "^8.5.2"

### ISSUE THREAD

**Issue by aradradev** [john-smilga/node-express-course#81]
In the controller/jobs.js -> showStats
When I try this code from the course it is not working for the new Mongoose version  "mongoose": "^8.5.2":
```js
const showStats = async (req, res) => {
  let stats = await Job.aggregate([
    { $match: { createdBy: new mongoose.Types.ObjectId(req.user.userId) } },
    { $group: { _id: '$status', count: { $sum: 1 } } },
  ])
  console.log(stats)
  res.status(StatusCodes.OK).json({ defaultStats: {}, monthlyApplications: [] })
}
```

This is the issue from my console:

![issueJobs](https://github.com/user-attachments/assets/2d87cf3a-3603-4db3-9336-306be3406710)

When you try the suggestion using the `new ` it will work but there will be some deprecation warnings:
![codeIssue](https://github.com/user-attachments/assets/8e41ae47-271f-4297-b6e6-d630a820c50d)

To solve the issue what I did is to use the string interpolation to make sure that I am passing a string.

![fixed](https://github.com/user-attachments/assets/14fcd601-0444-4c55-8976-01ad4fc33389)

Of course, if you're not using the new version of Node and Mongoose, this might not work for you.

**Comment by liamli99** [comment]
For the latest mongoose, please use `mongoose.Types.ObjectId.createFromHexString(req.user.userId)`!

**Comment by aradradev** [comment]
> For the latest mongoose, please use `mongoose.Types.ObjectId.createFromHexString(req.user.userId)`!

Yeah, this is another alternative, this might also work. However, using string interpolation worked for me.
